src/utils/dif_upload.rs

```diff
@@ -1518,8 +1514,16 @@ impl DifUpload {
         self.kinds.is_empty() || self.kinds.contains(&kind)
     }
 
-    /// Determines if this `ObjectClass` matches the search criteria.
-    fn valid_class(&self, class: ObjectClass) -> bool {
-        self.classes.is_empty() || self.classes.contains(&class)
+    /// Determines if the given `ObjectFeature`s match the search criteria.
+    fn valid_features(&self, features: &BTreeSet<ObjectFeature>) -> bool {
+        if features.is_empty() {
```

Comment:
This seems weird that looking for no features never returns matches rather than all.